### PR TITLE
✨  Pull request support for GitHub action

### PR DESCRIPTION
**Pull request by laurentsimon**
This adds support for pull request in the GH action.
No breaking changes.

The changes required updating how we generate our SARIF results.

This https://github.com/github/codeql-action/issues/811 is an improvement that would be nice, but non-blocking.

### CODE REVIEW

File: actions/Dockerfile [hunk 1]
```diff
@@ -12,9 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Testing: docker run -e INPUT_SARIF_FILE=results.sarif -e GITHUB_WORKSPACE=/ -e INPUT_POLICY_FILE="/policy.yml" -e INPUT_REPO_TOKEN=$GITHUB_AUTH_TOKEN -e GITHUB_REPOSITORY="ossf/scorecard" laurentsimon/scorecard-action:latest
+# Testing: docker run -e GITHUB_EVENT_NAME=pull_request -e INPUT_SARIF_FILE=results.sarif -e GITHUB_WORKSPACE=/ -e INPUT_POLICY_FILE="/policy.yml" -e INPUT_REPO_TOKEN=$GITHUB_AUTH_TOKEN -e GITHUB_REPOSITORY="ossf/scorecard" laurentsimon/scorecard-action:latest
```
Comment (naveensrinivasan) [inline actions/Dockerfile:15]: This is not part of this PR. But could we also please change the base image. This reduces the attack vector.

Comment (laurentsimon) [reply actions/Dockerfile:15]: Distroless images don't support shell. So to use distroless, we need to remove the shell script and create another golang project to call as a binary. Doable, but needs some work.
We also have this tracking issue https://github.com/ossf/scorecard/issues/1072

**Review by azeemshaikh38 — APPROVED** [review]

**Review by chrismcgehee — APPROVED** [review]
